ITISFoundation/osparc-simcore
# Is1273/groups api

## What do these changes do?

- new **catalog** service
   - [x] basic
   - [x] Add new table of dag to `simcore-postgres-database` package
- **webserver**: new catalog-gateway subsystem
  - [x] api exposed . Increased to 0.3.
  - [x] auth: requires login
  - [x] access: ``services.catalog.*`` permission required
  - [x] routing (reverse-proxy?)

#### Maintenance:
 - web API doc: documented and created tags in openapi 
 - makefile: webserver and catalog service makefile have all ``build`` and ``test`` target flavours
 - adds [shellcheck](https://www.shellcheck.net) tooling in ``scripts/shellcheck`` and vscode configuration in [.vscode-template/settings.json](.vscode-template/settings.json)
 - NEW standard makefile in ``simcore-postgres-database``
       -  highlight: has a setup to create migration commits
  - **IMPORTANT**: this PR requires upgrading deployed databases

## Related issue number
Implements #1273 

## How to test

- catalog service has some unit testing coverage
- catalog subsystem in webserver has some unit-testing converage
```cmd
cd services/catalog
make devel
make install-dev tests
```
- start stacks
```cmd
cd osparc-simcore
make devel
make build up-prod
```
Now open browser

- To see the webserver  OAS: http://127.0.0.1:9081/api/doc
![image](https://user-images.githubusercontent.com/32402063/74976622-5edaa080-5429-11ea-9d46-539215e3d13a.png)
- To see the catalog OAS: http://127.0.0.1:8005/docs or http://127.0.0.1:8005/redoc
![image](https://user-images.githubusercontent.com/32402063/74978291-939c2700-542c-11ea-9fae-a342673c30ec.png)
![image](https://user-images.githubusercontent.com/32402063/74978318-a1ea4300-542c-11ea-9db9-a2034da63b14.png)

## Checklist
